File: furniture-backend/src/main/java/com/example/furniture_backend/service/PaymentService.java
package com.example.furniture_backend.service;

import java.util.Map;

public interface PaymentService {

    /**
     * Create a Stripe payment intent
     * @param amount Amount in rupees
     * @return Map containing clientSecret
     */
    Map<String, String> createPaymentIntent(double amount);

    /**
     * Process payment (legacy method)
     */
    String processPayment(double amount);
}

File: furniture-frontend/src/services/paymentService.js
import api from "./api";

/**
 * Create a payment intent for Stripe
 * @param {number} amount - Amount in INR (rupees)
 * @returns {Promise<{clientSecret: string}>}
 */
export const createPaymentIntent = async (amount) => {
  const response = await api.post("/payment/create-intent", { amount });
  const data = response?.data;
  // Backend returns { clientSecret } or { error }
  if (data && typeof data === "object" && data.error) {
    return { error: data.error };
  }
  return data || {};
};